### DISCUSSION TITLE
Endpoint para políticos

### DISCUSSION THREAD

**Discussion by marvin-env** [BrasilAPI/BrasilAPI#224] [Ideas]
Já tem alguém trabalhando em um endpoint com uma lista de políticos e seus respectivos contatos públicos? Acho que isso seria bem legal e é uma contribuição que eu gostaria de trabalhar. Até já dei uma brincada com o puppeteer, acho que ficaria bem legal. 
![image](https://user-images.githubusercontent.com/14362970/98595588-eec0c300-22b4-11eb-8dbc-4b9eafad1096.png)

**Comment by alexandresanlim** [comment]
Poderia ter tbem o gasto mensal, usando o portal da transparência

**Comment by marvin-env** [comment]
> Poderia ter tbem o gasto mensal, usando o portal da transparência

Essa é uma ótima ideia.

**Comment by lucianopf** [comment]
Opa mestre @mcavalcante28  acho que essa ideia é novidade e curti bastante ! 😍 

Tu chegou a iniciar alguma implementação? 😍 🚀

**Comment by marvin-env** [comment]
@lucianopf 
Na verdade sim. Mas ainda estava primeiro validando pegar os dados da forma correta. 
Não tem nem um lugar que centraliza essa informações, estão espalhadas em vários sites do governo e então é necessário fazer um tratamento diferente. Acima eu tinha pego os dados dos senadores e mais tarde no mesmo dia eu fiz algo parecido pros nossos 513 deputados (foi meio trabalhoso, mas no final deu certo e foi bem divertido).

Resumo: estava caminhando, não fiz funcionar em forma de API ainda. Não fui mais adiante por enquanto, mas adoraria.
![image](https://user-images.githubusercontent.com/14362970/99259600-00e0bb00-27f9-11eb-929e-332ddd92ff46.png)

**Comment by lucianopf** [comment]
Que sensacional mestre! 🚀 🔝 

Se precisar de ajuda pra tirar isso do protótipo e transformar em PR me avisa que a gnt tenta fazer um bem bolado!
Parabéns pela iniciativa!! 👏 🎉

**Comment by marvin-env** [comment]
@lucianopf vou reservar um tempo pra estudar o Next e vou iniciar. Você terá notícias, pode deixar chefe! kk

**Comment by mklemersson** [comment]
@lucianopf @mcavalcante28 se for possível, dá pra usar o ranking dos políticos como base já que lá se tem a toda a listagem inclusive com os filtros para senado e camara dos deputados, talvez de pra falar com o pessoal de lá e fazer um proxy pra esse projeto.

**Comment by Gu7z** [comment]
Cara to so acompanhando o andamento e tive uma ideia. @mcavalcante28 consegue adicionar o email dos politicos? Seria muito bom ter essa informação!

**Comment by little-richard** [comment]
@mcavalcante28 , andei pesquisando sobre e a câmara fornece uma API Rest onde você consegue todas as informações do congresso, incluindo proposições votadas, gastos dos parlamentares e mais algumas informações interessantes. Deixo o link abaixo da API Rest. Também tenho interesse em ajudar e trabalhar com essa ferramenta. Gostaria de saber como está o desenvolvimento em relação a esses endpoints dos políticos para que eu possa ajudar vocês a implementar e pensar em algumas ideias. Desde já obrigado. 

dadosabertos.camara.leg.br/swagger/api.html

**Comment by alexandresanlim** [comment]
Também fico a disposição de colaborar.

**Comment by kopzinski** [comment]
@ramsantos muito legal essa API, certamente pode ser usada nesse contexto

**Comment by shuhikari** [comment]
Essa vai ser sensacional, poderíamos até acompanhar os processos nos quais os políticos estão envolvidos, centralizando todos os dados abertos disponíveis.  Isso também seria uma forma de combate às fake news, usando informações do que já foi protocolado no MP. 

Pode contar comigo.

**Comment by marvin-env** [comment]
Bom dia pessoal. 

Como tem muitas pessoas comentando com muitas ideias, gostaria de convidá-los a um servidor no discord, para que possamos marcar um dia para conversarmos e pra poder alinhar direitinho como será feito. 
Essa é a minha First Issue em contribuições open source, imagino que a diversão e a motivação de contribuir deve ser ter esse trabalho em equipe. Então quem quiser chegar junto entra aqui. 
https://discord.gg/cBHR5nXK8S

**Comment by joaopaulonsoares** [comment]
Boa Noite!!
Atualmente sou desenvolvedor no LABHacker da Câmara dos Deputados e possuo um certo conhecimento dessas API's dos dados abertos citados e de alguns metadados presentes nelas que por muitas vezes são bem confusos, acho muito interessante a ideia e estou disposto a contribuir o máximo que puder!

**Comment by joaopaulonsoares** [comment]
Inclusive todas nossas aplicações são voltadas para estimular a participação pública e feitas sempre com o código aberto. Caso queiram conhecer mais [https://github.com/labhackercd](https://github.com/labhackercd)

**Comment by alexandresanlim** [comment]
> Boa Noite!!
> Atualmente sou desenvolvedor no LABHacker da Câmara dos Deputados e possuo um certo conhecimento dessas API's dos dados abertos citados e de alguns metadados presentes nelas que por muitas vezes são bem confusos, acho muito interessante a ideia e estou disposto a contribuir o máximo que puder!

No caso dos gastos, uma vez eu estava estudando e via que as informações eram bem antigas (3 meses), e assim me desmotivou a criar algo, saberia o motivo?

**Comment by joaopaulonsoares** [comment]
> > Boa Noite!!
> > Atualmente sou desenvolvedor no LABHacker da Câmara dos Deputados e possuo um certo conhecimento dessas API's dos dados abertos citados e de alguns metadados presentes nelas que por muitas vezes são bem confusos, acho muito interessante a ideia e estou disposto a contribuir o máximo que puder!
> 
> No caso dos gastos, uma vez eu estava estudando e via que as informações eram bem antigas (3 meses), e assim me desmotivou a criar algo, saberia o motivo?

Saberia me informar quando foi isso? Durante essa pandemia a câmara passou por diversas atualizações e podem ter trocado o endpoint sem atualizar a documentação no momento. Saberia me dizer se nessa época as informações no site também estavam com esse período desatualizado?

**Comment by alexandresanlim** [comment]
> Saberia me informar quando foi isso? Durante essa pandemia a câmara passou por diversas atualizações e podem ter trocado o endpoint sem atualizar a documentação no momento. Saberia me dizer se nessa época as informações no site também estavam com esse período desatualizado?

Faz mto tempo, foi no meio do ano passado se não me engano.
Não cheguei a verificar as informações no site.

**Comment by allanhenriquers** [comment]
> @mcavalcante28 , andei pesquisando sobre e a câmara fornece uma API Rest onde você consegue todas as informações do congresso, incluindo proposições votadas, gastos dos parlamentares e mais algumas informações interessantes. Deixo o link abaixo da API Rest. Também tenho interesse em ajudar e trabalhar com essa ferramenta. Gostaria de saber como está o desenvolvimento em relação a esses endpoints dos políticos para que eu possa ajudar vocês a implementar e pensar em algumas ideias. Desde já obrigado.
> 
> dadosabertos.camara.leg.br/swagger/api.html

senhores existe um projeto que consome e trata os gastos dos políticos: [serenata de amor](https://github.com/okfn-brasil/serenata-de-amor), no que tange aos gastos seria interesante entender como eles fazem!
